notebooks/Notebook1.md

```diff
@@ -18,17 +24,35 @@ exciting milestones, preliminary results, etc. Feel free to include images
+
+https://docs.google.com/document/d/1yPfdPlC9sN5qvzx74-tXSt-7NBsu8DFBAY_Z_-y3AB8/edit?usp=sharing
+
+I have actually implemented all of the features on the first page of this document in Python (which I used to construct the Hearts program), but the features on the second page are the ones currently giving me trouble.
+
 ## Questions
 
 **What is the most pressing issue for your project? What design decision do
 you need to make, what implementation issue are you trying to solve, or how
 are you evaluating your design and implementation?**
 
+Far and away the most pressing issue for my project is the Round-Turn-Action-Move sequence. I feel confident that the Round implementation won't be terrible, and most of this week was spent getting Move to be as ironed out as possible. I'm just not totally certain that Scala has the right tools to make the Turn and Action sequences quite as crisp as I want them to be, but I think I'll be able to find something good enough; I just suspect it's going to take a lot of time. On the bright side, I think I've found a pretty good design loop, which I was using this week: I wrote a full program in my temporary Python DSL, then added features to that DSL to reduce the size of my program, shrinking it quite considerably (and making it much more readable). This seems like a good way to evaluate the strength of my DSL in general, although writing programs in it that serve as functional test cases is quite daunting at times, and may take up the bulk of my time in one of the later weeks.
+
 **What questions do you have for your critique partners? How can they best help
 you?**
 
+First, I want to know if my goal realignment makes sense for the language; is it really better that my language should be more fully-featured at the expense of its readability?
+Second, I have a design conundrum: I'm not sure whether to have a few classes with a lot of functionality, all of which will not necessarily be used in every program, or many classes with more specific functionality. I'm leaning towards the former, as it feels more intuitive: someone programming in the language is more likely to understand a Card class, rather than a CardWithTextAndFlavor class and a CardWithPowerAndToughness class. If they need more specific options, or need additional features, the language will still support subclassing and inheritance.
+
```

Comment:
If the different functionalities are independent, you might consider a "mixin" architecture, which allows you to write small classes, then mix in behavior that is defined in traits:

[https://alvinalexander.com/scala/how-to-use-scala-traits-as-mixins-mixing-in-traits/](https://alvinalexander.com/scala/how-to-use-scala-traits-as-mixins-mixing-in-traits/)

Let me know if you want to chat about this option.